src/Views/Homepage.vala

```diff
@@ -98,11 +99,62 @@ namespace AppCenter {
                 new Thread<void*> ("update-banner", run);
             });
 
+            var recently_updated_label = new Gtk.Label (_("Recently Updated"));
+            recently_updated_label.get_style_context ().add_class ("h4");
+            recently_updated_label.xalign = 0;
+            recently_updated_label.margin_start = 10;
+
+            var recently_updated_carousel = new Widgets.Carousel ();
+
+            var recently_updated_grid = new Gtk.Grid ();
+            recently_updated_grid.margin = 2;
+            recently_updated_grid.margin_top = 12;
+            recently_updated_grid.attach (recently_updated_label, 0, 0, 1, 1);
+            recently_updated_grid.attach (recently_updated_carousel, 0, 1, 1, 1);
+
+            var recently_updated_revealer = new Gtk.Revealer ();
+            recently_updated_revealer.add (recently_updated_grid );
+
+            houston.get_updated.begin ((obj, res) => {
+                var updated_ids = houston.get_updated.end (res);
+                List<AppCenterCore.Package> packages_for_carousel = new List<AppCenterCore.Package> ();
+                ThreadFunc<void*> run = () => {
+                    uint packages_added = 0;
+                    foreach (var package in updated_ids) {
+                        if (packages_added >= NUM_PACKAGES_IN_CAROUSEL) {
+                            break;
+                        }
+
+                        var candidate = package + ".desktop";
+                        var candidate_package = AppCenterCore.Client.get_default ().get_package_for_component_id (candidate);
+
+                        if (candidate_package != null) {
+                            candidate_package.update_state ();
+                            if (candidate_package.state == AppCenterCore.Package.State.NOT_INSTALLED) {
+                                packages_added++;
+                                packages_for_carousel.append (candidate_package);
+                            }
+                        }
+                    }
+                    
+                    Idle.add (() => {
+                        foreach (var banner_package in packages_for_carousel) {
+                            recently_updated_carousel.add_package (banner_package);
+                        }
+                        recently_updated_revealer.reveal_child = true;
+                        main_window.homepage_loaded ();
```

Comment:
We shouldn't be calling `main_window.homepage_loaded ()` twice. It's already called by the banner code. It causes a refresh of AppCenter's package cache which is slow and network/CPU intensive.

The reason why this exists is because the cache refresh was previously happening while we were trying to see if packages were installed. Because that was happening, we had to wait until that finished and it slowed the homepage load. So we deferred that until after the homepage was finished doing its stuff.

I imagine only calling it once in either the banner or the carousel will be fine. Otherwise, we need to come up with a more inventive solution to make sure it happens only once after both have loaded.